plugin/pkg/scheduler/schedulercache/cache.go

```diff
@@ -161,7 +161,7 @@ func (cache *schedulerCache) ForgetPod(pod *v1.Pod) error {
 	defer cache.mu.Unlock()
 
 	currState, ok := cache.podStates[key]
-	if currState.pod.Spec.NodeName != pod.Spec.NodeName {
+	if ok && currState.pod.Spec.NodeName != pod.Spec.NodeName {
```

Comment:
I'm OK with this code improvement, although did not get time for this case/logic :).